lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -9,23 +10,38 @@ class PokemonOverviewPage extends StatelessWidget {
     super.key,
   });
 
-  final List<Pokemon> pokemons;
+  final Async<List<Pokemon>> pokemons;
 
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      home: Scaffold(
-        appBar: AppBar(title: const Text(pokemonOverviewTitle)),
-        body: GridView.builder(
-          padding: const EdgeInsets.all(10.0),
-          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
-          itemCount: pokemons.length,
-          itemBuilder: (_, index) {
-            final pokemon = pokemons[index];
-            return PokemonTileCard(pokemon: pokemon);
-          },
+      home: Builder(
+        builder: (BuildContext context) => Scaffold(
+          appBar: AppBar(title: const Text(pokemonOverviewTitle)),
+          body: pokemons.when(
+            (data) => GridView.builder(
+              padding: const EdgeInsets.all(10.0),
+              gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
+              itemCount: data.length,
+              itemBuilder: (_, index) {
+                final pokemon = data[index];
+                return PokemonTileCard(pokemon: pokemon);
+              },
+            ),
+            loading: () => const Center(child: CircularProgressIndicator()),
+            error: (errorMessage) {
+              WidgetsBinding.instance.addPostFrameCallback((_) {
+                _showErrorMessageSnackbar(context, errorMessage);
+              });
+              return const Center(child: Text(emptyPokemonsLabel));
+            },
+          ),
         ),
       ),
     );
   }
+
+  _showErrorMessageSnackbar(BuildContext context, String? errorMessage) {
+    ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(errorMessage!)));
```

Comment:
use null coalescing and use `emptyString` constant as its fallback value instead of using non-null enforce here to be safer 🙂

also, don't forget to apply trailing comma here 🙂